shared/util/saga.js

```diff
@@ -23,20 +45,23 @@ function* sequentially(effects: Array<any>): Generator<any, Array<any>, any> {
 // TODO I couldn't get flow to figure out how to infer this, or even force you to explicitly do it
 // maybe flow-strict fixes this
 type MaybeAction = void | boolean | TypedActions | null
-function* chainAction<Actions>(
+function* chainAction<Actions: {+type: string}>(
```

Comment:
Does this really need to be covariant?

Reply:
It's because `type` is covariant in the generated actions